autoload/ale/lsp/message.vim

```diff
@@ -98,7 +102,7 @@ function! ale#lsp#message#Completion(buffer, line, column, trigger_character) ab
     \   'textDocument': {
     \       'uri': ale#path#ToURI(expand('#' . a:buffer . ':p')),
     \   },
-    \   'position': {'line': a:line - 1, 'character': a:column},
+    \   'position': {'line': a:line - 1, 'character': a:column - 1},
```

Comment:
I don't think this one works. I think for completion, you have to send the character after the character the cursor is on, so you tell the server you want to get a result like `adcdef` for `abc`, like so:

```
abc
   ^
```

We may what to decrement the column number here, but also increment it in `completion.vim`, which might be less confusing. I tried using completion with `rls` with and without this change, and it seems to work without this, and not with it.

Reply:
I think you are right, I've been testing with `asyncomplete-ale` and `asyncomplete` seems to be doing the right thing (supplying cur column + 1) in its context so it worked for me. Fixing `completion.vim` then.

Reply:
I went this the simplest/safest approach there, which fixes the position just for LSP, hope it's ok.